api/v1beta1/disruption_types.go

```diff
@@ -43,11 +43,12 @@ type DisruptionSpec struct {
 	// +nullable
 	AdvancedSelector []metav1.LabelSelectorRequirement `json:"advancedSelector,omitempty"` // advanced label selector
 	// +nullable
-	Filter          *DisruptionFilter `json:"filter,omitempty"`
-	DryRun          bool              `json:"dryRun,omitempty"`          // enable dry-run mode
-	OnInit          bool              `json:"onInit,omitempty"`          // enable disruption on init
-	Unsafemode      *UnsafemodeSpec   `json:"unsafeMode,omitempty"`      // unsafemode spec used to turn off safemode safety nets
-	StaticTargeting bool              `json:"staticTargeting,omitempty"` // enable dynamic targeting and cluster observation
+	Filter          *DisruptionFilter   `json:"filter,omitempty"`
+	DryRun          bool                `json:"dryRun,omitempty"`          // enable dry-run mode
+	OnInit          bool                `json:"onInit,omitempty"`          // enable disruption on init
+	Unsafemode      *UnsafemodeSpec     `json:"unsafeMode,omitempty"`      // unsafemode spec used to turn off safemode safety nets
+	StaticTargeting bool                `json:"staticTargeting,omitempty"` // enable dynamic targeting and cluster observation
+	Trigger         *DisruptionTriggers `json:"triggers,omitempty"`        // alter the pre-injection lifecycle
```

Comment:
So maybe let's keep Triggers for the field also?
```suggestion
	Triggers         *DisruptionTriggers `json:"triggers,omitempty"`        // alter the pre-injection lifecycle
```